src/main/java/mate/academy/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,75 @@
+import mate.academy.bookstore.dto.book.BookWithoutCategoryDto;
+import mate.academy.bookstore.dto.category.CategoryDto;
+import mate.academy.bookstore.dto.category.CategoryRequestDto;
+import mate.academy.bookstore.service.BookService;
+import mate.academy.bookstore.service.CategoryService;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category menegment", description = "Endpoints for managing categories")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @GetMapping
+    @Operation(summary = "Find all categories", description = "Get a list of all categories")
+    public List<CategoryDto> getAll() {
+        return categoryService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    @Operation(summary = "Get a category by id", description = "Get a category by id")
+    public CategoryDto getCategoryById(@PathVariable Long id) {
+        return categoryService.getById(id);
+    }
+
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create a new category", description = "Create a new category")
+    public CategoryDto createCategory(@RequestBody @Valid CategoryRequestDto requestDto) {
+        return categoryService.save(requestDto);
+    }
+
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/{id}")
+    @Operation(summary = "Delete a category by id", description = "Delete a category by id")
+    public void deleteCategory(@PathVariable Long id) {
+        categoryService.deleteById(id);
+    }
+
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @PutMapping("/{id}")
+    @Operation(summary = "Update category", description = "Update a category by id")
+    public CategoryDto updateCategory(@PathVariable Long id,
+                                      @RequestBody CategoryRequestDto requestDto) {
+        return categoryService.update(id, requestDto);
+    }
+
+    @GetMapping("/{id}/books")
+    @Operation(summary = "Get books by category",
+            description = "Get a list of books by category id")
+    public List<BookWithoutCategoryDto> getBooksByCategoryId(@PathVariable Long id) {
```

Comment:
```suggestion
    public List<BookWithoutCategoryDto> getBooksByCategoryId(@PathVariable @Positive Long id) {
```
Check other id 
Add `@Validated` under the controller